src/Evolution/Ringdown/MinimumAhCExcisionRadius.cpp

```diff
@@ -0,0 +1,362 @@
+          domain::creators::Sphere::Excision{nullptr},
+          static_cast<size_t>(0),
+          static_cast<size_t>(5),
+          false,
+          std::nullopt,
+          {50.0},
+          domain::CoordinateMaps::Distribution::Linear,
+          ShellWedges::All,
+          ringdown_time_dependent_map_options};
+
+      const auto temporary_ringdown_rmin_domain =
+          ringdown_rmin_domain_creator.create_domain();
+      const auto ringdown_rmin_functions_of_time =
+          ringdown_rmin_domain_creator.functions_of_time();
+
+      const auto exc_a_block_logical = block_logical_coordinates(
+          temporary_ringdown_rmin_domain, excision_a_inspiral_inertial_points,
+          match_time, ringdown_rmin_functions_of_time);
+      const auto exc_b_block_logical = block_logical_coordinates(
+          temporary_ringdown_rmin_domain, excision_b_inspiral_inertial_points,
+          match_time, ringdown_rmin_functions_of_time);
+
+      const tnsr::I<DataVector, 3, Frame::Inertial>
+          excision_a_ringdown_grid_points{
+              get<0>(ylm::cartesian_coords(excision_a_inspiral_grid)).size()};
+      const tnsr::I<DataVector, 3, Frame::Inertial>
+          excision_b_ringdown_grid_points{
+              get<0>(ylm::cartesian_coords(excision_b_inspiral_grid)).size()};
+
+      double min_excision_radius = 0.0;
+      tnsr::I<double, 3, Frame::Inertial> exc_a_inspiral_inertial_point{};
+      tnsr::I<double, 3, Frame::Grid> exc_a_ringdown_grid_point{};
+      tnsr::I<double, 3, Frame::Inertial> exc_b_inspiral_inertial_point{};
+      tnsr::I<double, 3, Frame::Grid> exc_b_ringdown_grid_point{};
+      for (size_t s = 0; s < get<0>(excision_a_inspiral_inertial_points).size();
+           ++s) {
+        get<0>(exc_a_inspiral_inertial_point) =
+            get<0>(excision_a_inspiral_inertial_points)[s];
+        get<1>(exc_a_inspiral_inertial_point) =
+            get<1>(excision_a_inspiral_inertial_points)[s];
+        get<2>(exc_a_inspiral_inertial_point) =
+            get<2>(excision_a_inspiral_inertial_points)[s];
+        get<0>(exc_b_inspiral_inertial_point) =
+            get<0>(excision_b_inspiral_inertial_points)[s];
+        get<1>(exc_b_inspiral_inertial_point) =
+            get<1>(excision_b_inspiral_inertial_points)[s];
+        get<2>(exc_b_inspiral_inertial_point) =
+            get<2>(excision_b_inspiral_inertial_points)[s];
+
+        // If the point is mapped to a block then the ringdown excision radius
+        // chosen does not enclose excisions A/B from the inspiral
+        if (exc_a_block_logical[s].has_value()) {
+          const auto& block_id_and_coords = exc_a_block_logical[s].value();
+          const auto& block = temporary_ringdown_rmin_domain
+                                  .blocks()[block_id_and_coords.id.get_index()];
+          const auto& grid_to_inertial_map =
+              block.moving_mesh_grid_to_inertial_map();
+          const auto inv_point = grid_to_inertial_map.inverse(
```

Comment:
Can't you use block-logical -> grid instead? Probably cheaper because it doesn't involve the shape map and it's guaranteed to succeed.